BuildMacros.cmake

```diff
@@ -1,4 +1,5 @@
 include(CMakeParseArguments)
+include(cmake/Sanitizers.cmake)
```

Comment:
Will this `include` pass even if the user is using a container without sanitizer support?

Reply:
I can't see any reason why it wouldn't but i tested it with `ldmx use dev v3.1` and it passes. Technically, the older versions of LDMX-sw already has all the support that is specified in this file, the dynamic libraries are just required at runtime to run binaries that were compiled with sanitizers enabled. 

However, this part does still raise a question for me which is what the proper way to include something like this in CMake  would be. This works, but I'm not confident enough in CMake to know if there is a more proper way to do it

Reply:
Scratch the part about the libraries, you of course need them to link...

Reply:
As long as the `include` doesn't fail, I'm fine with there being a linker error if a user requests the santizers but is not using the updated container image.

The "proper" way is to provide the path to the custom cmake modules and then load them using `include`. [We already provide the path to the `cmake` directory in the central `CMakeLists.txt`](https://github.com/LDMX-Software/ldmx-sw/blob/2d60422fd5d78a58b0b3d0004bf3fc1848a33cf8/CMakeLists.txt#L11-L12), so you can change your line to 
```cmake
include(Sanitizers)
```
which should work the same as what you have now. I guess this goes into my other comment, is there a way for you to check if the correct libraries exist within cmake so we could issue a helpful warning to the user who may have updated their ldmx-sw but not their container.

https://github.com/LDMX-Software/cmake/pull/15#discussion_r781428411

Reply:
Running `include(Sanitizers)` didn't work earlier and I've found out what the issue was. I wasn't doing `include(Sanitizers)`, I was doing `include(Sanitizers.cmake)`, which doesn't work (but `include(cmake/Sanitizers.cmake)` did!). I'll patch that in. 

Regarding the part of warning people. It looks like all of the parts required to compile, link, and run things with sanitizers enabled are actually already installed (I even went back to v2.0 and checked), it looks like they might just be distributed with (or is a dependency of) some part of the ubuntu parckages that we were using. So LDMX-sw built with e.g. v3.1 and turning on one of the sanitizers will actually both build, link, and run. You just won't get as good debug information though. 

In principle, it is definitely possible to issue some kind of warning but it would take a bit of work if you wanted to do it properly (i.e. just not checking if some particular file is available manually). Do we think that this would be worth the effort if older container still would work? 

Or maybe we could just check the version of the container being used and if it is older than 3.2 && the user is explicitly enabling one of the `-DENABLE_SANITIZER_...` option, we issue a warning then. 

This would also need to check so that it doesn't fire in a non-container world or a container without a specific version (such as a local container). Is there a way to get the container version (perhaps from `ldmx config`?)?

Reply:
If it compiles, links, and runs (just without as good debug info) then I don't think a warning is necessary. I was just worried about the case where someone tries to enable sanitizers and suddenly "everything breaks". Unless the better debug info is necessary in your opinion (I haven't used it), I think we are good to go.